### PR TITLE
TSERV-172: send custom properties in execution request

### PR DESCRIPTION
**Pull request by prakashmrec**

### CODE REVIEW

File: src/main/java/com/smartbear/readyapi/client/execution/CodegenBasedTestServerApi.java [hunk 1]
```diff
@@ -293,37 +296,60 @@ public void setBasePath(String basePath) {
     }
 
     @Override
-    public ProjectResultReport postProject(File file, boolean async, HttpBasicAuth auth, String testCaseName, String testSuiteName, String environment) throws ApiException {
```
Comment (olensmar) [inline src/main/java/com/smartbear/readyapi/client/execution/CodegenBasedTestServerApi.java]: we could keep and deprecate the old methods and simply delegate internally to the new ones?

File: src/main/java/com/smartbear/readyapi/client/execution/ProjectExecutionRequest.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        public Builder withProjectFile(File projectFile) {
+            projectExecutionRequest.projectFile = projectFile;
+            return this;
+        }
+
+        /**
+         * @param testSuiteName (optional) Name of the test suite in the project if specific test suite needs to be run.
+         * @return Builder
+         */
+        public Builder testSuite(String testSuiteName) {
```
Comment (olensmar) [inline src/main/java/com/smartbear/readyapi/client/execution/ProjectExecutionRequest.java]: perhaps forTestSuite and forTestCase ?

File: src/main/java/com/smartbear/readyapi/client/execution/ProjectExecutionRequest.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+         * @param environmentName (optional) Name of the environment (defined in project) if project has multiple environments defined and
+         *                        request is to execute for a particular environment
+         * @return Builder
+         */
+        public Builder forEnvironment(String environmentName) {
+            projectExecutionRequest.environment = environmentName;
+            return this;
+        }
+
+        public Builder addCustomProperty(String targetName, String propertyName, String value) {
```
Comment (olensmar) [inline src/main/java/com/smartbear/readyapi/client/execution/ProjectExecutionRequest.java]: withCustomProperty ?

File: src/main/java/com/smartbear/readyapi/client/execution/RecipeExecutor.java [hunk 4]
```diff
@@ -101,18 +95,14 @@ public Execution executeRepositoryProject(RepositoryProjectExecutionRequest exec
         return execution;
     }
 
-    public Execution submitProject(File project, @Nullable String testCaseName, @Nullable String testSuiteName, @Nullable String environment) throws ApiException {
```
Comment (olensmar) [inline src/main/java/com/smartbear/readyapi/client/execution/RecipeExecutor.java]: same as above - deprecate and delegate?

File: src/main/java/com/smartbear/readyapi/client/execution/RecipeExecutor.java [hunk 5]
```diff
@@ -101,18 +95,14 @@ public Execution executeRepositoryProject(RepositoryProjectExecutionRequest exec
...
+        Execution execution = doExecuteProject(projectExecutionRequest, true);
         notifyRequestSubmitted(execution);
         return execution;
     }
 
-    public Execution executeProject(File project) {
-        return executeProject(project, null, null, null);
-    }
-
-    public Execution executeProject(File project, @Nullable String testCaseName, @Nullable String testSuiteName, @Nullable String environment) throws ApiException {
```
Comment (olensmar) [inline src/main/java/com/smartbear/readyapi/client/execution/RecipeExecutor.java]: deprecate / delegate?

File: src/main/java/com/smartbear/readyapi/client/execution/RecipeExecutor.java [hunk 6]
```diff
@@ -194,9 +184,9 @@ private Execution doExecuteProjectFromRepository(RepositoryProjectExecutionReque
         }
     }
 
-    private Execution doExecuteProject(File project, boolean async, String testCaseName, String testSuiteName, String environment) throws ApiException {
```
Comment (olensmar) [inline src/main/java/com/smartbear/readyapi/client/execution/RecipeExecutor.java]: deprecate / delegate?

File: src/main/java/com/smartbear/readyapi/client/execution/TestServerApi.java [hunk 7]
```diff
@@ -15,7 +15,7 @@
  */
 public interface TestServerApi {
 
-    ProjectResultReport postProject(File file, boolean async, HttpBasicAuth auth, @Nullable String testCaseName, @Nullable String testSuiteName, @Nullable String environment) throws ApiException;
```
Comment (olensmar) [inline src/main/java/com/smartbear/readyapi/client/execution/TestServerApi.java]: deprecate / delegate?

File: src/main/java/com/smartbear/readyapi/client/execution/RecipeExecutor.java (hunk 6 above)
Comment (prakashmrec) [inline src/main/java/com/smartbear/readyapi/client/execution/RecipeExecutor.java]: Sure, I will fix.

File: src/main/java/com/smartbear/readyapi/client/execution/TestServerApi.java (hunk 7 above)
Comment (prakashmrec) [inline src/main/java/com/smartbear/readyapi/client/execution/TestServerApi.java]: Even for API interface and implementation? I think it's enough to keep old methods in RecipeExecutor.

Comment (olensmar) [inline src/main/java/com/smartbear/readyapi/client/execution/TestServerApi.java]: yes - sorry - you're right :-)

**Comment by olensmar** [comment]
thank you!

**Pull request merged**